File: dotnet/Lexy/Lexy.Compiler/Language/Expressions/Functions/YearsFunction.cs
using Lexy.Compiler.Parser;

namespace Lexy.Compiler.Language.Expressions.Functions
{
    public class YearsFunction : EndStartDateFunction
    {
        public const string Name = "YEARS";

        protected override string FunctionName => Name;

        private YearsFunction(Expression endDateExpression, Expression startDateExpression, SourceReference reference)
            : base(endDateExpression, startDateExpression, reference)
        {
        }

        public static BuiltInFunction Create(SourceReference reference, Expression endDateExpression,
            Expression startDateExpression) =>
            new YearsFunction(endDateExpression, startDateExpression, reference);
    }
}

File: typescipt/src/language/expressions/functions/yearsFunction.ts


export class YearsFunction extends EndStartDateFunction {
   public const string Name = `YEARS`;

   protected override string FunctionName => Name;

   constructor(endDateExpression: Expression, startDateExpression: Expression, reference: SourceReference)
     : base(endDateExpression, startDateExpression, reference) {
   }

   public static ExpressionFunction Create(SourceReference reference, Expression endDateExpression,
     Expression startDateExpression) {
     return new YearsFunction(endDateExpression, startDateExpression, reference);
   }
}
